x/auth/keeper/msg_server.go

```diff
@@ -20,6 +23,35 @@ func NewMsgServerImpl(ak AccountKeeper) types.MsgServer {
 	}
 }
 
+func (ms msgServer) NonAtomicExec(goCtx context.Context, msg *types.MsgNonAtomicExec) (*types.MsgNonAtomicExecResponse, error) {
+	if msg.Signer == "" {
+		return nil, errors.New("empty signer address string is not allowed")
+	}
+
+	signer, err := ms.ak.AddressCodec().StringToBytes(msg.Signer)
+	if err != nil {
+		return nil, sdkerrors.ErrInvalidAddress.Wrapf("invalid signer address: %s", err)
+	}
+
+	if len(msg.Msgs) == 0 {
+		return nil, sdkerrors.ErrInvalidRequest.Wrapf("messages cannot be empty")
+	}
+
+	msgs, err := msg.GetMessages()
```

Comment:
we're unpacking from any here and re-packing later – is it for validate basic?

Reply:
yes